### DISCUSSION TITLE
cut audio with specific start and end time?

### DISCUSSION THREAD

**Discussion by igormenin** [naudio/NAudio#1151] [Q&A]
Hello. I would like to know if with this library it would be possible for me to cut an audio starting from a specific time and ending a few seconds after I set it?  all of this running on a net 8 server on Debian Linux.

**Comment by markheath** [comment]
To trim an audio file seek the reader to your start (reader.CurrentTime = start) and copy until you reach the end time. 

On Linux, though, AudioFileReader/WaveFileReader handle WAV, but for MP3/FLAC/Ogg you'll want the cross-platform NAudio.SoundFile package (SoundFileReader, wraps libsndfile) — MediaFoundationReader is Windows-only. SoundFileReader is a WaveStream, so seek + OffsetSampleProvider/Take gives you the cut, and SoundFileWriter writes it back out. See [CrossPlatformAudioFilesWithSoundFile.md](https://github.com/naudio/NAudio/blob/main/Docs/CrossPlatformAudioFilesWithSoundFile.md).
